testsuite/src/test/java/org/ironjacamar/common/metadata/spec/JCA10TestCase.java

```diff
@@ -80,6 +81,104 @@ public void testRead() throws Exception
    }
 
    /**
+    * Write
+    * @throws Exception In case of an error
+    */
+   @Test
+   public void testWrite() throws Exception
+   {
+      RaParser parser = new RaParser();
+
+      InputStream is = JCA10TestCase.class.getClassLoader().
+         getResourceAsStream("../../resources/test/spec/ra-1.0.xml");
+      assertNotNull(is);
+
+      XMLInputFactory inputFactory = XMLInputFactory.newInstance();
+      inputFactory.setProperty(XMLInputFactory.SUPPORT_DTD, Boolean.FALSE);
+
+      XMLStreamReader xsr = inputFactory.createXMLStreamReader(is);
+      
+      Connector c = parser.parse(xsr);
+      assertNotNull(c);
+
+      is.close();
+
+      StringReader sr = new StringReader(c.toString());
+      XMLStreamReader nxsr = XMLInputFactory.newInstance().createXMLStreamReader(sr);
+      Connector cn = parser.parse(nxsr);
+      checkConnector(cn);
+      assertEquals(c, cn);
+   }
+
+   /**
+    * Equal
+    * @throws Exception In case of an error
+    */
+   @Test
+   public void testEqual() throws Exception
+   {
+      RaParser parser = new RaParser();
+
+      InputStream is1 = JCA10TestCase.class.getClassLoader().
+         getResourceAsStream("../../resources/test/spec/ra-1.0.xml");
+      assertNotNull(is1);
+
+      XMLInputFactory inputFactory = XMLInputFactory.newInstance();
+      inputFactory.setProperty(XMLInputFactory.SUPPORT_DTD, Boolean.FALSE);
+
+      XMLStreamReader xsr1 = inputFactory.createXMLStreamReader(is1);
+      
+      Connector c1 = parser.parse(xsr1);
+      assertNotNull(c1);
+
+      is1.close();
+
+      InputStream is2 = JCA10TestCase.class.getClassLoader().
+         getResourceAsStream("../../resources/test/spec/ra-1.0.xml");
+      assertNotNull(is2);
+
+      XMLInputFactory inputFactory2 = XMLInputFactory.newInstance();
+      inputFactory2.setProperty(XMLInputFactory.SUPPORT_DTD, Boolean.FALSE);
+
+      XMLStreamReader xsr2 = inputFactory2.createXMLStreamReader(is2);
+      
+      Connector c2 = parser.parse(xsr2);
+      assertNotNull(c2);
+
+      is2.close();
+
+      assertEquals(c1, c2);
+   }
+
+   /**
+    * Copy
+    * @throws Exception In case of an error
+    */
+   @Test
+   public void testCopy() throws Exception
+   {
+      RaParser parser = new RaParser();
+
+      InputStream is1 = JCA10TestCase.class.getClassLoader().
+         getResourceAsStream("../../resources/test/spec/ra-1.0.xml");
+      assertNotNull(is1);
+
+      XMLInputFactory inputFactory = XMLInputFactory.newInstance();
+      inputFactory.setProperty(XMLInputFactory.SUPPORT_DTD, Boolean.FALSE);
+
+      XMLStreamReader xsr1 = inputFactory.createXMLStreamReader(is1);
+      
+      Connector c1 = parser.parse(xsr1);
+      assertNotNull(c1);
+
+      is1.close();
+
+      Connector c2 = c1.copy();
+      
+      assertEquals(c1, c2);
```

Comment:
I think for good measure also assert that c1 != c2 on a pointer level, and their toString's are the same